### ISSUE TITLE
enable pushState on browser history

### ISSUE THREAD

**Issue by farskid** [bencompton/framework7-react#43]
The original pure JS framework7 has a `pushState` property in it's initiation configurations. It enables the browser history control and synced routing with URl. In the framework7-react docs, It's written that all those configuration props are supporter. Although I use `pushState` as a prop on `Framework7App` component, It doesn't enable this feature.

```javascript
export const App = () => (
  <Framework7App pushState={true} routes={routes}>
    <MainView />
  </Framework7App>
)
```

**Comment by renatoholanda** [comment]
I have the same problem. This prop will show correct url when access pages, right?

**Comment by farskid** [comment]
Right. for example if you set a `/about/` path and __About__ component on routes, by visiting `/about/`, It gotta mount __About__ component and update the window.location.href.

**Comment by renatoholanda** [comment]
Yep. I did think that is my mistake. Kkkk So, let's wait the dev answer.

**Comment by renatoholanda** [comment]
I've tested here and works it. What's your version?

**Comment by farskid** [comment]
1.6.2

**Comment by renatoholanda** [comment]
1.6.2? But the last version is 0.8.5.

**Comment by farskid** [comment]
I thought you are referring to framework7 version. `0.8.5-beta.10`

**Comment by renatoholanda** [comment]
I'm sorry. You said about BROWSER HISTORY. That not works here.

**Comment by farskid** [comment]
I can't even get my hashHistory work.

**Comment by bencompton** [comment]
@farskid - hash history should be working. For example, if I go into the `Framework7App` component in the kitchen sink and add a prop for `pushState={true}` just like you did, I can navigate around the kitchen sink and see the URL change in the browser.

Does it work in the kitchen sink for you? If so, perhaps you could share what your `<MainView />` component looks like?

If you get into more advanced use cases beyond hash history, you'll likely run into some trouble. The router in Framework7 Vue still needs some work in order to be able to support all of the use cases that React Router and Vue Router do. You can see that there are several open issues about the router / push state in Framework7 Vue:  https://github.com/nolimits4web/Framework7-Vue. Framework7 React basically just converts the Vue components into React components, so the same issues exist in Framework7 React.

**Comment by farskid** [comment]
@bencompton Could you please tell me where I can find framewrok7-react kitchen sink?

**Comment by bencompton** [comment]
Check out these instructions: 

https://github.com/bencompton/framework7-react#kitchen-sink.

Alternatively, if you don't already have the kitchen sink running, it might be quicker for you to clone a starter template and set `pushState={true}`:

https://github.com/bencompton/framework7-react-app-template

It should work equally well in either.
